IPython/core/display.py

```diff
@@ -173,19 +176,108 @@ def display(*objs, include=None, exclude=None, metadata=None, transient=None, di
         If given as True, generate a new display_id
     kwargs: additional keyword-args, optional
         Additional keyword-arguments are passed through to the display publisher.
-    
+
     Returns
     -------
-    
+
     handle: DisplayHandle
         Returns a handle on updatable displays, if display_id is given.
         Returns None if no display_id is given (default).
+
+    Examples
+    --------
+
+    >>> class Json(object):
+    ...     def __init__(self, json):
+    ...         self.json = json
+    ...     def _repr_pretty_(self, pp, cycle):
+    ...         import json
+    ...         pp.text(json.dumps(self.json, indent=2))
+    ...     def __repr__(self):
+    ...         return str(self.json)
+    ...
+
+    >>> d = Json({1:2, 3: {4:5}})
+
+    >>> print(d)
+    {1: 2, 3: {4: 5}}
+
+    >>> display(d)
+    {
+      "1": 2,
+      "3": {
+        "4": 5
+      }
+    }
+
+    >>> def int_formatter(integer, pp. clycle):
+    ...     pp.text('I'*integer)
+
+    >>> plain = get_ipython().display_formatter.formatters['text/plain']
+    >>> plain.for_type(int, int_formatter)
+    >>> display(7 - 5)
+    II
+
+    >>> del plain.type_printers[int]
+    >>> display(3+6)
+    9
```

Comment:
Totally minor nitpick, but it'd be even more clear if the display calls both used the same expression inside... 
Make both `display(3+6)` for example, so the top one is even more clear about how it's doing the formatting...